makefiles/features_check.inc.mk

```diff
@@ -3,9 +3,17 @@
 # Features that are used without taking "one out of" dependencies into account
 FEATURES_USED_SO_FAR := $(sort $(FEATURES_REQUIRED) $(FEATURES_OPTIONAL_USED))
 
-_features_would_conflict := $(if $(findstring $1,$2),$(filter-out $1,$(subst :, ,$2)))
-FEATURES_WOULD_CONFLICT := $(foreach features,$(FEATURES_USED_SO_FAR),\
-                             $(call _features_would_conflict,$(features),$(FEATURES_CONFLICT)))
+# Get features which inclusion would cause a conflict
+# Parameter 1: Features currently used
+# Parameter 2: A set of features that would conflict (separated by spaces)
+# Algorithm: If interaction of the two lists is empty, return an empty set. Otherwise return
+#            the set of conflicting features without the feature in it that is already used
+_features_would_conflict := $(if $(filter $1,$2),$(filter-out $1,$($2)))
```

Comment:
I don't quite get the `$($2)` here.

Reply:
That's a leftover. I wonder, why this works. Let me double check if the test is working.

Reply:
Aha! Because I placed the unit tests in the wrong folder and it never get's run :/

Reply:
Btw: Functions (for use with `$(call)` cannot be simply expended variables, the have to be recursively expended variables. Because otherwise they won't store the "function source code", but rather the "functions return value" at the time they are defined.